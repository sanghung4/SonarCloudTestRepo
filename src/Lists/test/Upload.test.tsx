import { fireEvent, waitFor } from '@testing-library/react';

import userEvent from '@testing-library/user-event';

import { render, renderWithComponent } from 'test-utils/TestWrapper';
import { setBreakpoint } from 'test-utils/mockMediaQuery';

import ListsProvider from 'providers/ListsProvider';
import ListUpload from 'Lists/Upload';
import {
  ACCEPTED_FILES,
  BASE_USE_DROPZONE,
  CSV_FILE,
  FILE_REJECTIONS,
  JSON_FILE,
  mockData
} from './Upload.mocks';
import * as t from 'locales/en/translation.json';
import { useDropzone } from 'react-dropzone';

window.HTMLElement.prototype.scrollIntoView = jest.fn();

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn()
  }),
  useLocation: () => ({
    pathname: '/lists/upload'
  })
}));

jest.mock('react-dropzone', () => ({
  useDropzone: jest.fn()
}));

const ACCOUNTS = {
  selectedAccounts: {
    billTo: {
      id: 'testAccount'
    },
    shipTo: {
      id: 'testAccount'
    }
  }
};

describe('List Upload tests', () => {
  beforeAll(() => setBreakpoint('desktop'));
  beforeEach(() => {
    (useDropzone as jest.Mock).mockReturnValue(BASE_USE_DROPZONE);
  });

  it('Should display a link to download the file upload template file', async () => {
    const { container, queryByTestId } = render(
      <ListsProvider>
        <ListUpload />
      </ListsProvider>,
      {
        selectedAccountsConfig: ACCOUNTS
      }
    );

    await waitFor(() => {
      expect(queryByTestId('download-template-link')).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  it('Should add a CSV file through the file picker', async () => {
    const { getByTestId, queryByTestId } = render(
      <ListsProvider>
        <ListUpload />
      </ListsProvider>,
      {
        selectedAccountsConfig: ACCOUNTS
      }
    );

    await waitFor(() => {
      expect(queryByTestId('choose-file-button')).toBeInTheDocument();
    });

    const dropzone = getByTestId('dropzone-root');
    const input = getByTestId('choose-file-input') as HTMLInputElement;

    userEvent.upload(input, CSV_FILE, undefined, { applyAccept: true });

    expect(input.files?.[0]).toStrictEqual(CSV_FILE);
    expect(input.files?.item(0)).toStrictEqual(CSV_FILE);
    expect(input.files).toHaveLength(1);
    expect(dropzone).toMatchSnapshot();
  });

  it('Should add a CSV file with drag and drop', async () => {
    (useDropzone as jest.Mock).mockReturnValue({
      ...BASE_USE_DROPZONE,
      ...ACCEPTED_FILES
    });

    const {
      render: { getByTestId, queryByTestId, queryByText, rerender },
      component
    } = renderWithComponent(
      <ListsProvider>
        <ListUpload />
      </ListsProvider>,
      {
        selectedAccountsConfig: ACCOUNTS
      }
    );
    const data = mockData([CSV_FILE]);

    await waitFor(() => {
      expect(queryByTestId('choose-file-button')).toBeInTheDocument();
    });

    const dropzone = getByTestId('dropzone-root');
    const input = getByTestId('choose-file-input') as HTMLInputElement;

    fireEvent.drop(dropzone, data);
    fireEvent.change(input, {
      target: {
        files: [CSV_FILE]
      }
    });

    await waitFor(() => rerender(component));
    await waitFor(() => {
      expect(
        queryByText(`${t.lists.fileSelected}: ${CSV_FILE.name}`)
      ).toBeInTheDocument();
    });
    expect(dropzone).toMatchSnapshot();
  });

  it('Should not add a file besides a CSV through the file picker', async () => {
    (useDropzone as jest.Mock).mockReturnValue({
      ...BASE_USE_DROPZONE,
      ...FILE_REJECTIONS
    });

    const { getByTestId, queryByTestId, queryByText } = render(
      <ListsProvider>
        <ListUpload />
      </ListsProvider>,
      {
        selectedAccountsConfig: ACCOUNTS
      }
    );

    await waitFor(() => {
      expect(queryByTestId('choose-file-button')).toBeInTheDocument();
    });

    const dropzone = getByTestId('dropzone-root');
    const input = getByTestId('choose-file-input') as HTMLInputElement;

    userEvent.upload(input, JSON_FILE, undefined, { applyAccept: true });

    expect(
      queryByText(`${t.lists.fileSelected}: ${JSON_FILE.name}`)
    ).not.toBeInTheDocument();

    await waitFor(() => {
      expect(queryByTestId('upload-csv-error')).toBeInTheDocument();
    });
    expect(dropzone).toMatchSnapshot();
  });

  it('Should not add a file besides a CSV with drag and drop', async () => {
    (useDropzone as jest.Mock).mockReturnValue({
      ...BASE_USE_DROPZONE,
      ...FILE_REJECTIONS
    });

    const {
      render: { getByTestId, queryByTestId, queryByText, rerender },
      component
    } = renderWithComponent(
      <ListsProvider>
        <ListUpload />
      </ListsProvider>,
      {
        selectedAccountsConfig: ACCOUNTS
      }
    );
    const data = mockData([JSON_FILE]);

    await waitFor(() => {
      expect(queryByTestId('choose-file-button')).toBeInTheDocument();
    });

    const dropzone = getByTestId('dropzone-root');
    const input = getByTestId('choose-file-input') as HTMLInputElement;

    fireEvent.drop(dropzone, data);
    fireEvent.change(input, {
      target: {
        files: [JSON_FILE]
      }
    });

    await waitFor(() => rerender(component));
    expect(
      queryByText(`${t.lists.fileSelected}: ${JSON_FILE.name}`)
    ).not.toBeInTheDocument();
    expect(dropzone).toMatchSnapshot();

    await waitFor(() => {
      expect(queryByTestId('upload-csv-error')).toBeInTheDocument();
    });
    expect(dropzone).toMatchSnapshot();
  });

  it('Should remove an uploaded file', async () => {
    const { getByTestId, queryByTestId } = render(
      <ListsProvider>
        <ListUpload />
      </ListsProvider>,
      {
        selectedAccountsConfig: ACCOUNTS
      }
    );

    await waitFor(() => {
      expect(queryByTestId('choose-file-button')).toBeInTheDocument();
    });

    const dropzone = getByTestId('dropzone-root');
    const input = getByTestId('choose-file-input') as HTMLInputElement;

    userEvent.upload(input, CSV_FILE, undefined, { applyAccept: true });

    const fileNameLabel = getByTestId('file-name-label');
    expect(fileNameLabel).toBeInTheDocument();
    expect(fileNameLabel).toHaveTextContent(CSV_FILE.name);
    expect(input.files?.length).toEqual(1);
    expect(dropzone).toMatchSnapshot('ADDED');

    const removeFileButton = getByTestId('remove-file-button');
    userEvent.click(removeFileButton);
    expect(dropzone).toMatchSnapshot('REMOVED');
  });
});

describe('List Upload tests - Mobile', () => {
  beforeAll(() => setBreakpoint('mobile'));
  beforeEach(() => {
    (useDropzone as jest.Mock).mockReturnValue(BASE_USE_DROPZONE);
  });

  it('Should not allow drag-and-drop', async () => {
    const { getByTestId, queryByTestId, queryByText } = render(
      <ListsProvider>
        <ListUpload />
      </ListsProvider>,
      {
        selectedAccountsConfig: ACCOUNTS
      }
    );

    await waitFor(() => {
      expect(queryByTestId('choose-file-button')).toBeInTheDocument();
    });

    await waitFor(() => {
      expect(queryByText(t.lists.dragDrop)).not.toBeInTheDocument();
    });

    expect(getByTestId('dropzone-root')).toMatchSnapshot();
  });
});
