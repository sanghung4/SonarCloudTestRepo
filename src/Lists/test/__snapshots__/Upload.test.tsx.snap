// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Upload tests - Mobile Should not allow drag-and-drop 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation8 MuiCard-root css-cqq017-MuiPaper-root-MuiCard-root"
  data-testid="dropzone-root"
>
  <svg
    class="MuiBox-root css-i3pbo"
    sx="[object Object]"
  >
    cloud-upload.svg
  </svg>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1egkuzh-MuiButtonBase-root-MuiButton-root"
    data-testid="choose-file-button"
    tabindex="0"
    type="button"
  >
    Choose File
  </button>
  <input
    data-testid="choose-file-input"
    id="choose-file-form"
    name="chooseFileForm"
    style="display: none;"
    type="file"
  />
  <p
    class="MuiTypography-root MuiTypography-body1 css-1j9uqhc-MuiTypography-root"
  >
    File types: .csv
  </p>
</div>
`;

exports[`List Upload tests Should add a CSV file through the file picker 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation8 MuiCard-root css-17wkss1-MuiPaper-root-MuiCard-root"
  data-testid="dropzone-root"
>
  <form
    class="MuiBox-root css-1mi3tt8"
  >
    <div
      class="MuiBox-root css-13ce5na"
    >
      <div
        class="MuiBox-root css-1f4xe1w"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-5ds2jq-MuiTypography-root"
          data-testid="file-name-label"
        >
          File selected: list.csv
        </p>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root css-1dpl3kd-MuiButtonBase-root-MuiButton-root"
          data-testid="remove-file-button"
          tabindex="0"
          type="button"
        >
          Remove
        </button>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1vhyr2u-MuiFormControl-root-MuiTextField-root"
        data-testid="create-new"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-1411gyj-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="mui-1"
          id="mui-1-label"
        >
          Create a New List
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-tg1mzk-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1mjnin0-MuiInputBase-input-MuiOutlinedInput-input"
            data-testid="upload-new-list-input"
            id="mui-1"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-m1dym3-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1z7n62"
            >
              <span>
                Create a New List
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 css-wbviwo-MuiTypography-root"
      >
        or
      </p>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-l6ikmy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Select Existing List
      </button>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root Mui-disabled  css-1iol619-MuiButtonBase-root-MuiButton-root"
        disabled=""
        tabindex="-1"
        type="submit"
      >
        Upload File
      </button>
    </div>
  </form>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1j9uqhc-MuiTypography-root"
  >
    File types: .csv
  </p>
</div>
`;

exports[`List Upload tests Should add a CSV file with drag and drop 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation8 MuiCard-root css-17wkss1-MuiPaper-root-MuiCard-root"
  data-testid="dropzone-root"
>
  <form
    class="MuiBox-root css-1mi3tt8"
  >
    <div
      class="MuiBox-root css-13ce5na"
    >
      <div
        class="MuiBox-root css-1f4xe1w"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-5ds2jq-MuiTypography-root"
          data-testid="file-name-label"
        >
          File selected: list.csv
        </p>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root css-1dpl3kd-MuiButtonBase-root-MuiButton-root"
          data-testid="remove-file-button"
          tabindex="0"
          type="button"
        >
          Remove
        </button>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1vhyr2u-MuiFormControl-root-MuiTextField-root"
        data-testid="create-new"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-1411gyj-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="mui-2"
          id="mui-2-label"
        >
          Create a New List
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-tg1mzk-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1mjnin0-MuiInputBase-input-MuiOutlinedInput-input"
            data-testid="upload-new-list-input"
            id="mui-2"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-m1dym3-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1z7n62"
            >
              <span>
                Create a New List
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 css-wbviwo-MuiTypography-root"
      >
        or
      </p>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-l6ikmy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Select Existing List
      </button>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root Mui-disabled  css-1iol619-MuiButtonBase-root-MuiButton-root"
        disabled=""
        tabindex="-1"
        type="submit"
      >
        Upload File
      </button>
    </div>
  </form>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1j9uqhc-MuiTypography-root"
  >
    File types: .csv
  </p>
</div>
`;

exports[`List Upload tests Should display a link to download the file upload template file 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1kbjjd5-MuiContainer-root"
  >
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1yp6lty-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-n1sla7-MuiBreadcrumbs-separator"
        >
          <svg>
            chevron-right.svg
          </svg>
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-10eimqc-MuiTypography-root-MuiLink-root"
            data-testid="my-lists-breadcrumb"
            to="[object Object]"
          >
            My Lists
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-n1sla7-MuiBreadcrumbs-separator"
        >
          <svg>
            chevron-right.svg
          </svg>
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-ohe2uh-MuiTypography-root"
            data-testid="file-upload-breadcrumb"
          >
            File Upload
          </span>
        </li>
      </ol>
    </nav>
  </div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-cnms5y-MuiContainer-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-gzzm4s-MuiPaper-root-MuiCard-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-16xaac9-MuiTypography-root"
        data-testid="file-upload-title"
      >
        File Upload
      </h5>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-118ksc7-MuiPaper-root-MuiCard-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-11e32h3-MuiTypography-root"
        data-testid="how-to-format-header"
      >
        How to Format a File
      </h6>
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1m6pi2w-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-c89szk-MuiTypography-root"
          >
            To upload a new list from your computer, first
             
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1a0g8bi-MuiTypography-root-MuiLink-root"
              data-testid="download-template-link"
              download=""
              href="/templates/list_template.csv"
            >
              download our CSV template
            </a>
            .
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
          >
            Format your file
          </p>
          <div
            class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
            data-testid="upload-list-steps"
          >
            <ol
              class="css-1ddppw8"
            >
              <li>
                Keep column headers as they are formatted in the template
              </li>
              <li>
                Enter a Reece Part # in the Part # column (required)
              </li>
              <li>
                Enter product descriptions or product names in the Description column (optional)
              </li>
              <li>
                Enter the manufacturer or brand name in the MFR Name column (optional)
              </li>
              <li>
                Enter the quantity desired in the QTY column (required)
              </li>
              <li>
                Save the file as a .csv file
              </li>
            </ol>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1ym4c5j-MuiTypography-root"
          >
            Tip: Lists have a maximum capacity of 600 line items, for the best experience please do not upload list greater than 600 items.
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1m6pi2w-MuiGrid-root"
          data-testid="example-file"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1n0j5p9-MuiTypography-root"
          >
            Example file
            :
          </p>
          <img
            alt="Upload List Example File"
            class="MuiBox-root css-7ndvsj"
            src="Upload_List_Example_File.svg"
          />
        </div>
      </div>
      <h6
        class="MuiTypography-root MuiTypography-h6 css-1izcbvx-MuiTypography-root"
        data-testid="upload-the-file-header"
      >
        Upload the file
      </h6>
      <div
        class="MuiBox-root css-4dys60"
      >
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-2ae64n-MuiPaper-root-MuiAlert-root"
                data-testid="list-upload-alert"
                role="alert"
              >
                <div
                  class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
                >
                  <svg
                    class="MuiBox-root css-12z0wuy"
                    sx="[object Object]"
                  >
                    warning.svg
                  </svg>
                </div>
                <div
                  class="MuiAlert-message css-acap47-MuiAlert-message"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2 css-1wolbp5-MuiTypography-root"
                    data-testid="error-title"
                  >
                    Errors: 
                  </h6>
                  List could not be uploaded. Please correct the errors below and try again.
                </div>
                <div
                  class="MuiAlert-action css-ki1hdl-MuiAlert-action"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-ff60md-MuiButtonBase-root-MuiIconButton-root"
                    data-testid="lost-upload-close-alert"
                    tabindex="0"
                    type="button"
                  >
                    <svg>
                      delete.svg
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation8 MuiCard-root css-17wkss1-MuiPaper-root-MuiCard-root"
        data-testid="dropzone-root"
      >
        <svg
          class="MuiBox-root css-i3pbo"
          sx="[object Object]"
        >
          cloud-upload.svg
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body1 css-5ds2jq-MuiTypography-root"
          data-testid="drag-drop-text"
        >
          Drag and drop file here
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-o7buyu-MuiTypography-root"
        >
          or
        </p>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1egkuzh-MuiButtonBase-root-MuiButton-root"
          data-testid="choose-file-button"
          tabindex="0"
          type="button"
        >
          Choose File
        </button>
        <input
          data-testid="choose-file-input"
          id="choose-file-form"
          name="chooseFileForm"
          style="display: none;"
          type="file"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 css-1j9uqhc-MuiTypography-root"
        >
          File types: .csv
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`List Upload tests Should not add a file besides a CSV through the file picker 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation8 MuiCard-root css-17wkss1-MuiPaper-root-MuiCard-root"
  data-testid="dropzone-root"
>
  <svg
    class="MuiBox-root css-i3pbo"
    sx="[object Object]"
  >
    cloud-upload.svg
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-5ds2jq-MuiTypography-root"
    data-testid="drag-drop-text"
  >
    Drag and drop file here
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-o7buyu-MuiTypography-root"
  >
    or
  </p>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1egkuzh-MuiButtonBase-root-MuiButton-root"
    data-testid="choose-file-button"
    tabindex="0"
    type="button"
  >
    Choose File
  </button>
  <input
    data-testid="choose-file-input"
    id="choose-file-form"
    name="chooseFileForm"
    style="display: none;"
    type="file"
  />
  <p
    class="MuiTypography-root MuiTypography-body1 css-1aietki-MuiTypography-root"
    data-testid="upload-csv-error"
  >
    <svg
      class="MuiBox-root css-12z0wuy"
      sx="[object Object]"
    >
      warning.svg
    </svg>
    Error: File must be .CSV
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1j9uqhc-MuiTypography-root"
  >
    File types: .csv
  </p>
</div>
`;

exports[`List Upload tests Should not add a file besides a CSV with drag and drop 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation8 MuiCard-root css-17wkss1-MuiPaper-root-MuiCard-root"
  data-testid="dropzone-root"
>
  <svg
    class="MuiBox-root css-i3pbo"
    sx="[object Object]"
  >
    cloud-upload.svg
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-5ds2jq-MuiTypography-root"
    data-testid="drag-drop-text"
  >
    Drag and drop file here
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-o7buyu-MuiTypography-root"
  >
    or
  </p>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1egkuzh-MuiButtonBase-root-MuiButton-root"
    data-testid="choose-file-button"
    tabindex="0"
    type="button"
  >
    Choose File
  </button>
  <input
    data-testid="choose-file-input"
    id="choose-file-form"
    name="chooseFileForm"
    style="display: none;"
    type="file"
  />
  <p
    class="MuiTypography-root MuiTypography-body1 css-1aietki-MuiTypography-root"
    data-testid="upload-csv-error"
  >
    <svg
      class="MuiBox-root css-12z0wuy"
      sx="[object Object]"
    >
      warning.svg
    </svg>
    Error: File must be .CSV
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1j9uqhc-MuiTypography-root"
  >
    File types: .csv
  </p>
</div>
`;

exports[`List Upload tests Should not add a file besides a CSV with drag and drop 2`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation8 MuiCard-root css-17wkss1-MuiPaper-root-MuiCard-root"
  data-testid="dropzone-root"
>
  <svg
    class="MuiBox-root css-i3pbo"
    sx="[object Object]"
  >
    cloud-upload.svg
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-5ds2jq-MuiTypography-root"
    data-testid="drag-drop-text"
  >
    Drag and drop file here
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-o7buyu-MuiTypography-root"
  >
    or
  </p>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1egkuzh-MuiButtonBase-root-MuiButton-root"
    data-testid="choose-file-button"
    tabindex="0"
    type="button"
  >
    Choose File
  </button>
  <input
    data-testid="choose-file-input"
    id="choose-file-form"
    name="chooseFileForm"
    style="display: none;"
    type="file"
  />
  <p
    class="MuiTypography-root MuiTypography-body1 css-1aietki-MuiTypography-root"
    data-testid="upload-csv-error"
  >
    <svg
      class="MuiBox-root css-12z0wuy"
      sx="[object Object]"
    >
      warning.svg
    </svg>
    Error: File must be .CSV
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1j9uqhc-MuiTypography-root"
  >
    File types: .csv
  </p>
</div>
`;

exports[`List Upload tests Should remove an uploaded file: ADDED 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation8 MuiCard-root css-17wkss1-MuiPaper-root-MuiCard-root"
  data-testid="dropzone-root"
>
  <form
    class="MuiBox-root css-1mi3tt8"
  >
    <div
      class="MuiBox-root css-13ce5na"
    >
      <div
        class="MuiBox-root css-1f4xe1w"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-5ds2jq-MuiTypography-root"
          data-testid="file-name-label"
        >
          File selected: list.csv
        </p>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root css-1dpl3kd-MuiButtonBase-root-MuiButton-root"
          data-testid="remove-file-button"
          tabindex="0"
          type="button"
        >
          Remove
        </button>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1vhyr2u-MuiFormControl-root-MuiTextField-root"
        data-testid="create-new"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-1411gyj-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="mui-3"
          id="mui-3-label"
        >
          Create a New List
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-tg1mzk-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1mjnin0-MuiInputBase-input-MuiOutlinedInput-input"
            data-testid="upload-new-list-input"
            id="mui-3"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-m1dym3-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1z7n62"
            >
              <span>
                Create a New List
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 css-wbviwo-MuiTypography-root"
      >
        or
      </p>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-l6ikmy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Select Existing List
      </button>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root Mui-disabled  css-1iol619-MuiButtonBase-root-MuiButton-root"
        disabled=""
        tabindex="-1"
        type="submit"
      >
        Upload File
      </button>
    </div>
  </form>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1j9uqhc-MuiTypography-root"
  >
    File types: .csv
  </p>
</div>
`;

exports[`List Upload tests Should remove an uploaded file: REMOVED 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation8 MuiCard-root css-17wkss1-MuiPaper-root-MuiCard-root"
  data-testid="dropzone-root"
>
  <svg
    class="MuiBox-root css-i3pbo"
    sx="[object Object]"
  >
    cloud-upload.svg
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-5ds2jq-MuiTypography-root"
    data-testid="drag-drop-text"
  >
    Drag and drop file here
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-o7buyu-MuiTypography-root"
  >
    or
  </p>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1egkuzh-MuiButtonBase-root-MuiButton-root"
    data-testid="choose-file-button"
    tabindex="0"
    type="button"
  >
    Choose File
  </button>
  <input
    data-testid="choose-file-input"
    id="choose-file-form"
    name="chooseFileForm"
    style="display: none;"
    type="file"
  />
  <p
    class="MuiTypography-root MuiTypography-body1 css-1j9uqhc-MuiTypography-root"
  >
    File types: .csv
  </p>
</div>
`;
